`ipfs-unixfs-importer`:  `import` type declaration imports `UnixFS` from the wrong location

## Summary
`ipfs-unixfs-importer`:  `import` type declaration imports `UnixFS` from the wrong location

## Impact
unable to use `ipfs-unixfs-importer` package in typescript projects. attempting to transpile typescript project using `tsc` fails with the following error:

```shell
> tsc

node_modules/ipfs-unixfs-importer/types/src/index.d.ts:26:20 - error TS2307: Cannot find module 'ipfs-unixfs/src' or its corresponding type declarations.

26     unixfs: import("ipfs-unixfs/src").UnixFS | undefined;
                      ~~~~~~~~~~~~~~~~~


Found 1 error.
```

## Potential Cause
- `typesVersions` in [`package.json`](https://github.com/ipfs/js-ipfs-unixfs/blob/master/packages/ipfs-unixfs/package.json#L23-L34) of `ipfs-unixfs` has a path mapping that checks both `types/*` and `types/src/*`. Removing `typesVersions` entirely and regenerating the type declarations yields `unixfs: import("ipfs-unixfs").UnixFS | undefined` which solves the problem because typescript resolves the type using the `types` property in `package.json`. Though, i'm not sure if this is the way we want to solve the problem. I'm not sure whether downstream projects are importing types that aren't exported from `index.d.ts`. I think ideally, using something like [`npm-dts`](https://github.com/vytenisu/npm-dts) to bundle all types into 1 `index.d.ts` that is made available at the root of the package tarball would remove a lot of complexity around configuring types 

## Somewhat related
- type declaration files are duplicated in the package
![image](https://user-images.githubusercontent.com/4887440/169102657-3748abb8-a59a-432c-a398-ae85e411af0a.png)
- I think that the [`reset`](https://github.com/ipfs/js-ipfs-unixfs/blob/master/package.json#L20)  npm script needs to delete the `types` directory in every package because it's auto-generated. not deleting them prevents types from being regenerated

--- a/package.json
+++ b/package.json
@@ -17,7 +17,7 @@
   },
   "private": true,
   "scripts": {
-    "reset": "lerna run clean && rimraf ./node_modules ./package-lock.json packages/*/node_modules packages/*/package-lock.json packages/*/dist",
+    "reset": "lerna run clean && rimraf ./node_modules ./package-lock.json packages/*/node_modules packages/*/package-lock.json",
     "test": "lerna run --concurrency 1 test -- --",
     "test:node": "lerna run --concurrency 1 test:node -- --",
     "test:chrome": "lerna run --concurrency 1 test:chrome -- --",
@@ -26,6 +26,7 @@
     "test:firefox-webworker": "lerna run --concurrency 1 test:firefox-webworker -- --",
     "test:electron-main": "lerna run --concurrency 1 test:electron-main -- --",
     "test:electron-renderer": "lerna run --concurrency 1 test:electron-renderer -- --",
+    "clean": "lerna run clean",
     "build": "lerna run build",
     "lint": "lerna run lint",
     "dep-check": "lerna run dep-check",

--- a/packages/ipfs-unixfs-exporter/package.json
+++ b/packages/ipfs-unixfs-exporter/package.json
@@ -20,18 +20,6 @@
   },
   "type": "module",
   "types": "types/src/index.d.ts",
-  "typesVersions": {
-    "*": {
-      "*": [
-        "types/*",
-        "types/src/*"
-      ],
-      "types/*": [
-        "types/*",
-        "types/src/*"
-      ]
-    }
-  },
   "files": [
     "*",
     "!**/*.tsbuildinfo",
@@ -144,7 +132,7 @@
     "test:firefox": "aegir test -t browser -- --browser firefox",
     "build": "aegir build && cp -R types dist",
     "preleaseOnly": "npx json -I -f dist/package.json -e this.types='\"src/index.d.ts\"'",
-    "clean": "rimraf ./dist",
+    "clean": "rimraf ./dist ./types",
     "lint": "aegir ts -p check && aegir lint",
     "coverage": "nyc -s npm run test -t node && nyc report --reporter=html",
     "dep-check": "aegir dep-check -i @types/mocha -i @types/sinon -i nyc -i abort-controller -i rimraf -i copy -i util -i crypto-browserify -i events -i readable-stream -i interface-blockstore",

--- a/packages/ipfs-unixfs-importer/package.json
+++ b/packages/ipfs-unixfs-importer/package.json
@@ -20,18 +20,6 @@
   },
   "type": "module",
   "types": "types/src/index.d.ts",
-  "typesVersions": {
-    "*": {
-      "*": [
-        "types/*",
-        "types/src/*"
-      ],
-      "types/*": [
-        "types/*",
-        "types/src/*"
-      ]
-    }
-  },
   "files": [
     "*",
     "!**/*.tsbuildinfo",
@@ -144,7 +132,7 @@
     "test:firefox": "aegir test -t browser -- --browser firefox",
     "build": "aegir build && cp -R types dist",
     "preleaseOnly": "npx json -I -f dist/package.json -e this.types='\"src/index.d.ts\"'",
-    "clean": "rimraf ./dist",
+    "clean": "rimraf ./dist ./types",
     "lint": "aegir ts -p check && aegir lint",
     "coverage": "nyc -s npm run test -t node && nyc report --reporter=html",
     "de-pcheck": "aegir dep-check -i @types/mocha -i nyc -i rimraf -i copy -i util -i crypto-browserify -i events -i readable-stream -i assert",

--- a/packages/ipfs-unixfs/package.json
+++ b/packages/ipfs-unixfs/package.json
@@ -20,18 +20,6 @@
   },
   "type": "module",
   "types": "types/src/index.d.ts",
-  "typesVersions": {
-    "*": {
-      "*": [
-        "types/*",
-        "types/src/*"
-      ],
-      "types/*": [
-        "types/*",
-        "types/src/*"
-      ]
-    }
-  },
   "files": [
     "*",
     "!**/*.tsbuildinfo",
@@ -148,7 +136,7 @@
     "test:chrome": "aegir test -t browser --cov",
     "test:firefox": "aegir test -t browser -- --browser firefox",
     "build": "aegir build && cp -R types dist",
-    "clean": "rimraf ./dist",
+    "clean": "rimraf ./dist ./types",
     "lint": "aegir ts -p check && aegir lint",
     "coverage": "nyc -s aegir test -t node && nyc report --reporter=html",
     "dep-check": "aegir dep-check -i mkdirp -i @types/mocha -i nyc -i npm-run-all -i copy -i util",

